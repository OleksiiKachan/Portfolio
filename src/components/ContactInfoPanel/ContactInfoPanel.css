.contactInfoContainer {
    padding: 25px;
}

.contactInfo__title {
    display: block;
    margin-bottom: 10px;
}

.contactInfo__contactItem {
    margin: 30px 0;
	display: flex;
	align-items: center;
}

.contactInfo__icon {
    margin-right: 25px;
}