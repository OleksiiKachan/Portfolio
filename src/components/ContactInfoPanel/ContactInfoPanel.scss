.contactInfoContainer {
    padding: 25px;
}

.contactInfo__title {
    display: block;
    margin-bottom: 10px;
}