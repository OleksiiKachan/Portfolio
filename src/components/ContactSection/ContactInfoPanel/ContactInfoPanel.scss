.contactInfoPanel {
  padding: 25px;

  &__title {
    display: block;
    margin-bottom: 10px;
  }
}
