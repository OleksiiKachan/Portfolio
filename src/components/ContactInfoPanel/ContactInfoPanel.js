import React, { Component } from 'react';
import classNames from 'classnames';
import './ContactInfoPanel.css';
import PhoneIcon from './phone.svg';
import EmailIcon from './email.svg';
import LocationIcon from './location.svg';

/*
    <ContactInfoPanel contact={} />
*/
class ContactInfoPanel extends Component {
    render() {
        return (
            <div className={classNames('contactInfoContainer')}>
                <h3 className={classNames('contactInfo__title')}>Contacts</h3>
                <div className={classNames('contactInfo__contactItem')}>
                    <img className={classNames('contactInfo__icon')} src={PhoneIcon} alt='phone' />
                    <p className={classNames('contactInfo__text')}>{this.props.contacts.phone}</p>
                </div>
                <div className={classNames('contactInfo__contactItem')}>
                    <img className={classNames('contactInfo__icon')} src={EmailIcon} alt='email' />
                    <p className={classNames('contactInfo__text')}>{this.props.contacts.email}</p>
                </div>
                <div className={classNames('contactInfo__contactItem')}>
                    <img className={classNames('contactInfo__icon')} src={LocationIcon} alt='location' />
                    <p className={classNames('contactInfo__text')}>{this.props.contacts.location}</p>
                </div>
            </div>
        );
    }
}

export default ContactInfoPanel;