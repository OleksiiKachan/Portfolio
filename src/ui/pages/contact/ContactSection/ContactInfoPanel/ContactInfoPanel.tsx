import React from 'react';
import classNames from 'classnames';
import ContactInfoItem from '../ContactInfoItem/ContactInfoItem';
import './ContactInfoPanel.scss';

const icon_phone = require('../../../../assets/icons/phone.svg');
const icon_email = require('../../../../assets/icons/email.svg');
const icon_location = require('../../../../assets/icons/location.svg');

/*
    <ContactInfoPanel
      contact={}
      className=''
    />
*/

type PropsType = {
  contacts: {
    phone: { caption: string; url: string };
    email: { caption: string; url: string };
    location: { caption: string; url: string };
  };
  className?: string;
};

export default ({ contacts, className }: PropsType) => {
  const contactsData = [
    {
      icon: icon_phone,
      iconAlt: 'Phone icon',
      content: contacts.phone,
    },
    {
      icon: icon_email,
      iconAlt: 'Envelope icon',
      content: contacts.email,
    },
    {
      icon: icon_location,
      iconAlt: 'Compass icon',
      content: contacts.location,
    },
  ];
  return (
    <address className={classNames('contactInfoPanel', className)}>
      <h3 className={classNames('contactInfoPanel__title')}>Contacts</h3>
      {contactsData.map(contact => {
        return (
          <ContactInfoItem
            key={contactsData.indexOf(contact)}
            icon={contact.icon}
            iconAlt={contact.iconAlt}
            content={contact.content}
          />
        );
      })}
    </address>
  );
};
