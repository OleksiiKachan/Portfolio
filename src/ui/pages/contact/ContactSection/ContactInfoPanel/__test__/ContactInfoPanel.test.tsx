import React from 'react';
import { shallow } from 'enzyme';
import ContactInfoPanel from '../ContactInfoPanel';

describe('ContactInfoPanel', () => {
  it('Snapshot', () => {
    const component = shallow(
      <ContactInfoPanel
        contacts={{
          phone: 'phone number',
          email: 'email',
          location: 'location',
        }}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
