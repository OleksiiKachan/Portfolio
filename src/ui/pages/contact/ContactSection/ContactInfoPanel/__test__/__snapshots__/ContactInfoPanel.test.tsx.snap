// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfoPanel Snapshot 1`] = `
<div
  className="contactInfoPanel"
>
  <h3
    className="contactInfoPanel__title"
  >
    Contacts
  </h3>
  <_default
    content="phone number"
    icon={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "phone.svg",
      }
    }
    iconAlt="Phone icon"
    key="0"
  />
  <_default
    content="email"
    icon={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "email.svg",
      }
    }
    iconAlt="Envelope icon"
    key="1"
  />
  <_default
    content="location"
    icon={
      Object {
        "ReactComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "__esModule": true,
        "default": "location.svg",
      }
    }
    iconAlt="Compass icon"
    key="2"
  />
</div>
`;
